BaragonService/src/main/java/com/hubspot/baragon/service/worker/BaragonRequestWorker.java

```diff
@@ -245,36 +252,65 @@ public void run() {
+              .or(false)) {
+            upstreamOnlyRequests.add(request);
+            added++;
+          } else {
+            break;
+          }
+        }
+      }
+
+      List<QueuedRequestWithState> batchedUpstreamOnlyUpdates = upstreamOnlyRequests.stream()
+          .map(this::hydrateQueuedRequestWithState)
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .collect(Collectors.toList());
+
+      LOG.debug("Immediately sending out {} upstream-only updates", batchedUpstreamOnlyUpdates.size());
+
+      // These are batched requests which only update upstreams. Send these out immediately.
+      handleQueuedRequests(batchedUpstreamOnlyUpdates);
+
+      // Now handle the remaining requests as usual.
+      queuedRequestIds.removeAll(upstreamOnlyRequests);
+
       if (!queuedRequestIds.isEmpty()) {
         final Set<String> handledServices = Sets.newHashSet();
         final Set<QueuedRequestWithState> queuedRequestsWithState = Sets.newHashSet();
         for (QueuedRequestId queuedRequestId : queuedRequestIds) {
-          if (!handledServices.contains(queuedRequestId.getServiceId())) {
-            final String requestId = queuedRequestId.getRequestId();
-            final Optional<InternalRequestStates> maybeState = requestManager.getRequestState(requestId);
-
-            if (!maybeState.isPresent()) {
-              LOG.warn(String.format("%s does not have a request status!", requestId));
-              continue;
-            }
+          if (added >= configuration.getWorkerConfiguration().getMaxRequestsPerPoll()) {
+            break;
+          }
 
-            final Optional<BaragonRequest> maybeRequest = requestManager.getRequest(requestId);
+          if (!handledServices.contains(queuedRequestId.getServiceId())) {
+            Optional<QueuedRequestWithState> queuedRequestWithState = hydrateQueuedRequestWithState(queuedRequestId);
 
-            if (!maybeRequest.isPresent()) {
-              LOG.warn(String.format("%s does not have a request object!", requestId));
+            if (!queuedRequestWithState.isPresent()) {
               continue;
             }
 
-            queuedRequestsWithState.add(new QueuedRequestWithState(queuedRequestId, maybeRequest.get(), maybeState.get()));
+            queuedRequestsWithState.add(queuedRequestWithState.get());
             handledServices.add(queuedRequestId.getServiceId());
             added++;
           }
-          if (added >= configuration.getWorkerConfiguration().getMaxRequestsPerPoll()) {
-            break;
-          }
         }
 
-        Map<QueuedRequestWithState, InternalRequestStates> results = handleQueuedRequests(queuedRequestsWithState);
+        List<QueuedRequestWithState> prioritizedQueuedRequestsWithState = queuedRequestsWithState.stream().sorted(queuedRequestComparator()).collect(Collectors.toList());
```

Comment:
is this ordering still valid when there is a case like this in the queue?
- noreload/validate + doesn't change properties
- thing that changes properties
- another noreload/validate with the new properties from ^^

Trying to remember if we'd ever get something like that from singularity. But in this case we'd possibly skip validate on something that had updated properties